sipa/blueprints/generic.py

```diff
@@ -162,7 +165,18 @@ def logout():
                 extra={'tags': {'user': current_user.uid}})
     logout_user()
     flash(gettext("Abmeldung erfolgreich!"), 'success')
-    return redirect(url_for('.index'))
+
+    resp = redirect(url_for('.index'))
+
+    cookie_name = current_app.config.get('REMEMBER_COOKIE_NAME', COOKIE_NAME)
+    if cookie_name in request.cookies:
+        resp.set_cookie(cookie_name, "", 0)
+
+    current_app.login_manager._clear_cookie(resp)
+    resp.set_cookie("session", "")
+    session.clear()
+
+    return resp
```

Comment:
@sgeisler that's not the point. The question is: why delete the session cookie?

You are deleting something that is thought of as a basic feature flask provides. Other points of  the application (CSRF Tokens, for instance) might use it, but you say “fuck you, you're a cookie” and delete it.

The informations stored in the session are arbitrary things since this is a signed, base64-encoded cookie, and _not_ a session-id or similiar. Thus the session cannot be used “worse“ to track a user than with plain text cookies.

If it is really required to warn about these contents, then rather implement a cookie warning than breaking features.